extra_geom/detectors.py

```diff
@@ -803,41 +804,41 @@ def __init__(self, pixel_size: float, slow_pixels: int, fast_pixels: int,
         `tile_vec`: the direction of tile replication, default=[1, 0, 0]
 
         """
-        self.pixel_size = pixel_size
-        self.frag_fs_pixels = fast_pixels
-        self.frag_ss_pixels = slow_pixels
-        self.n_modules = len(corner_coordinates)
-        self.n_tiles_per_module = n_tiles_per_module
-        self.tile_gap = tile_gap if tile_gap else pixel_size
-        self.tile_vec = np.array(tile_vec) if tile_vec else ss_vec
+        cls.pixel_size = pixel_size
+        cls.frag_fs_pixels = fast_pixels
+        cls.frag_ss_pixels = slow_pixels
+        cls.n_modules = len(corner_coordinates)
+        cls.n_tiles_per_module = n_tiles_per_module
+        cls.tile_gap = tile_gap if tile_gap else pixel_size
+        cls.tile_vec = np.array(tile_vec) if tile_vec else ss_vec
```

Comment:
This will change the attributes on the class, rather than on an instance. So it will seem to work at first, but if you do:

```python
g1 = GenericGeometry.from_simple_description(pixel_size=2e-4, ...)
g2 = GenericGeometry.from_simple_description(pixel_size=1e-4, ...)
```

Then both `g1.pixel_size` and `g2.pixel_size` will be 1e-4 afterwards (because it's set `GenericGeometry.pixel_size`).

I think we want to calculate the list of modules, then create an instance and attach the attributes to that:

```python
...
inst = cls(modules)
inst.pixel_size = pixel_size
...
return inst
```

Reply:
Damn right! I've never thought about such a case, thanks! Will rework later today.

Reply:
Done. It looks a little clumsy, but seems to work.